Add logger

davishmcclurg opened the pull request:
This allows passing in a logger to write to an application's log file.

enkessler commented:
@davishmcclurg 

1. I like that you've got the logger outputting at various severity levels depending on what was triggering the output lines that you are replacing.
2. I like there is a default logger set up so that it only takes additional effort on a user's part if they want to use a custom one.

However,

1. The generator class seems to only be a local development tool and not for end user use, so I wouldn't bother wiring up the logger in there. Particularly because that is the only place that information was being output to `$stdout` instead of `$stderr` and I would prefer that...
2. ...the default is to log to `$stderr`, because that is where info was dumped before and that would keep things the most backwards compatible.

If you are so inclined, adding tests for new functionality is always appreciated so that I don't have to get around to doing it myself. ;)

davishmcclurg commented:
@enkessler I removed the logger from the generator class and spec helper. You're right that it doesn't make sense to use it there. I also changed the default to `$stderr` and added a couple specs. I had to switch to a `logger` class method to make the specs easier to write.

enkessler commented:
@davishmcclurg I've merged this into the development branch. It will go out with the next release.

Thanks!
